kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.delta.kernel.*;
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.types.*;
+
+import io.delta.kernel.internal.actions.*;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.replay.LogReplay;
+import io.delta.kernel.internal.snapshot.*;
+import io.delta.kernel.internal.util.SchemaUtils;
+import io.delta.kernel.internal.util.Tuple2;
+import static io.delta.kernel.internal.TransactionImpl.DEFAULT_READ_VERSION;
+import static io.delta.kernel.internal.TransactionImpl.DEFAULT_WRITE_VERSION;
+import static io.delta.kernel.internal.util.VectorUtils.stringArrayValue;
+import static io.delta.kernel.internal.util.VectorUtils.stringStringMapValue;
+
+public class TransactionBuilderImpl implements TransactionBuilder {
+    private static final Logger logger = LoggerFactory.getLogger(TransactionBuilderImpl.class);
+
+    private final long currentTimeMillis = System.currentTimeMillis();
+    private final TableImpl table;
+    private final String engineInfo;
+    private final Operation operation;
+    private Optional<StructType> schema = Optional.empty();
+    private Optional<List<String>> partitionColumns = Optional.empty();
+
+    public TransactionBuilderImpl(TableImpl table, String engineInfo, Operation operation) {
+        this.table = table;
+        this.engineInfo = engineInfo;
```

Comment:
I forget if you answered this somewhere else, but who is responsible for injecting the _kernel_ engine info? Or where does that happen?

Reply:
Added code when creating the commit info action to prepend "Kernel/"